File: lessons/compiled_lang/cpp/src/b00.variables.cpp
#include <iostream>
#include <string>

int main() {
    int number = 5;
    char character = 'a';
    std::string text = "This is a string";

    std::string output = "Number is " + std::to_string(number) + ".\n"
        + "Character is '" + character + "'.\n"
        + "String is \"" + text + "\".\n";

    std::cout << output;

    return 0;
}

File: lessons/compiled_lang/rust/src/b00.variables.rs
fn main() {
    let number = 5;
    let character = 'a';
    let text = "This is a string";

    let number_str = number.to_string();
    let character_str = character.to_string();

    let output = "Number is ".to_string()
        + &number_str + ".\n"
        + "Character is '" + &character_str + "'.\n"
        + "String is \"" + text + "\".\n";

    print!("{output}");
}
